miniconf_derive/src/lib.rs

```diff
@@ -182,60 +170,40 @@ fn derive_struct(
         impl #impl_generics miniconf::Miniconf for #ident #ty_generics #where_clause {
             fn set_path<'a, 'b: 'a, P, D>(&mut self, path_parts: &mut P, de: D) -> Result<(), miniconf::Error>
             where
-                P: miniconf::Peekable<Item = &'a str>,
+                P: Iterator<Item = &'a str>,
                 D: serde::Deserializer<'b>,
             {
-                let field = path_parts.next().ok_or(miniconf::Error::PathTooShort)?;
-                let peek = path_parts.peek().is_some();
-
-                match field {
+                match path_parts.next().ok_or(miniconf::Error::PathTooShort)? {
                     #(#set_path_arms ,)*
-                    _ => Err(miniconf::Error::PathNotFound)
+                    _ => Err(miniconf::Error::PathNotFound),
                 }
             }
 
             fn get_path<'a, P, S>(&self, path_parts: &mut P, ser: S) -> Result<S::Ok, miniconf::Error>
             where
-                P: miniconf::Peekable<Item = &'a str>,
+                P: Iterator<Item = &'a str>,
                 S: serde::Serializer,
             {
-                let field = path_parts.next().ok_or(miniconf::Error::PathTooShort)?;
-                let peek = path_parts.peek().is_some();
-
-                match field {
+                match path_parts.next().ok_or(miniconf::Error::PathTooShort)? {
                     #(#get_path_arms ,)*
                     _ => Err(miniconf::Error::PathNotFound)
                 }
             }
 
-            fn next_path<const TS: usize>(
-                state: &mut [usize],
-                path: &mut miniconf::heapless::String<TS>,
+            fn next_path(
+                state: &[usize],
+                depth: usize,
+                mut path: impl core::fmt::Write,
                 separator: char,
-            ) -> Result<bool, miniconf::IterError> {
-                let original_length = path.len();
-                loop {
-                    match *state.first().ok_or(miniconf::IterError::PathDepth)? {
-                        #(#next_path_arms ,)*
-                        _ => return Ok(false),
-                    };
-
-                    // Note(unreachable) Without any deferred fields, every arm above returns
-                    #[allow(unreachable_code)]
-                    {
-                        // If a deferred field is done, strip off the field name again,
-                        // and advance to the next field.
-                        path.truncate(original_length);
-
-                        state[0] += 1;
-                        state[1..].fill(0);
-                    }
+            ) -> Result<usize, miniconf::IterError> {
```

Comment:
Not really following what this return value is supposed to be. Is it in the docs of the trait now?

Reply:
I see now, the trait got updated.